Portal/vip-api/src/main/java/fr/insalyon/creatis/vip/api/business/DataApiBusiness.java

```diff
@@ -0,0 +1,608 @@
+        String lfcPath = getLFCPathFromApiUri(uploadData.getUri());
+        java.nio.file.Path javaPath = Paths.get(lfcPath);
+        String parentLfcPath = javaPath.getParent().toString();
+        checkPermission(lfcPath, AccessType.UPLOAD);
+        // check if parent dir exists
+        if (!baseDoesFileExist(parentLfcPath)) {
+            logger.error("parent directory of upload does not exist :" + lfcPath);
+            throw new ApiException("Upload Directory doest not exist");
+        }
+        // TODO : support archive upload
+        String uploadDirectory = DataManagerUtil.getUploadRootDirectory(false);
+        logger.debug("LFC path of new upload :" + lfcPath);
+        // TODO : normalize file name to avoid weird characters
+        String fileName = Paths.get(lfcPath).getFileName().toString();
+        String localPath = uploadDirectory + fileName;
+        logger.debug("storing upload file in :" + localPath);
+        // transform base64 string to a new file
+        writeFileFromBase64(uploadData.getPathContent(), localPath);
+        // transfer it
+        String operationId = baseUploadFile(localPath, parentLfcPath);
+        logger.info("File upload asked. Operation id : " + operationId);
+        // get new path
+        Path newPath = new Path();
+        newPath.setPlatformURI(uploadData.getUri());
+        newPath.setIsDirectory(false);
+        newPath.setMimeType(getMimeType(lfcPath));
+        newPath.setExists(false); // not created yet because asynchronous
+        return newPath;
+    }
+
+    public Path mkdir(String uri) throws ApiException {
+        String lfcPath = getLFCPathFromApiUri(uri);
+        java.nio.file.Path javaPath = Paths.get(lfcPath);
+        String parentLfcPath = javaPath.getParent().toString();
+        checkPermission(lfcPath, AccessType.UPLOAD);
+        // check if parent dir exists
+        if (!baseDoesFileExist(parentLfcPath)) {
+            logger.error("parent directory of upload does not exist :" + lfcPath);
+            throw new ApiException("Upload Directory doest not exist");
+        }
+        if (baseDoesFileExist(lfcPath)) {
+            logger.error("Trying do create an existing directory :" + lfcPath);
+            throw new ApiException("Upload error");
+        }
+        baseMkdir(parentLfcPath, javaPath.getFileName().toString());
+        Path newPath = new Path();
+        newPath.setPlatformURI(uri);
+        newPath.setIsDirectory(true);
+        newPath.setMimeType(env.getProperty(CarminProperties.API_DIRECTORY_MIME_TYPE));
+        newPath.setSize(0L);
+        newPath.setExists(true);
+        return newPath;
+    }
+
+    // #### PERMISSION STUFF
+
+    private enum AccessType {READ, UPLOAD, DELETE}
+
+    private void checkReadPermission(String lfcPath) throws ApiException {
+        checkPermission(lfcPath, AccessType.READ);
+    }
+
+    private void checkPermission(String lfcPath, AccessType accessType) throws ApiException {
```

Comment:
The thing is that currently, in VIP data-manager, the verification is done client-side, so no factorisation could be done (or it will imply a major intern refactoring of the data manager code).
Perhaps you also say that this code would have more sense to be in the data-manager module than in the API module, and I'm OK to move it.